File: hanghae-cinema/domain/src/main/java/org/example/domain/Movie.java
package org.example.domain;

import lombok.Builder;
import lombok.Getter;
import org.example.enums.AgeRating;
import org.example.enums.Genre;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Builder
public class Movie {
    private Long id;
    private String title;
    private Genre genre;
    private LocalDate releaseDate;
    private Integer runtimeMinutes;
    private AgeRating ageRating;
    private String thumbnailUrl;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: movie/core/src/main/kotlin/com/example/movie/domain/movie/model/Movie.kt
package com.example.movie.domain.movie.model

import com.example.movie.domain.common.model.BaseModel
import com.example.movie.domain.genre.model.Genre
import com.example.movie.domain.screening.model.Screening
import java.time.LocalDate
import java.time.LocalDateTime

class Movie(
    val id: Long,
    val title: String,
    val rating: Rating,
    val genre: Genre,
    val releaseDate: LocalDate,
    val thumbnailUrl: String,
    val runningTime: Int,
    val screenings: List<Screening>,
    override val createdBy: String,
    override val createdAt: LocalDateTime,
    override val updatedBy: String,
    override val updatedAt: LocalDateTime
) : BaseModel